File: server/ERP.DataService.Model/Model/Menu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace ERP.DataService.Model.Model
{
    public class Menu
    {
        public Guid Id { set; get; }
        [StringLength(50)]
        public string Name { set; get; }
        public Module module { set; get; }
         [StringLength(50)]
        public string RouterPath { set; get; }
        public string ImagePath { set; get; }
    }
}

File: Client/erp-client/src/app/models/admin/menu.ts
import { Submenu } from "./submenu";

export class Menu {
   public Id?:string;
   public Name:string;
   public RouterPath:string;
   public ImagePath:string;  
   public MenuSqenceId?:number;
   public ModuleSeqId?:number;
   public SubMenus:Submenu[]
}
